pkg/apis/crd.projectcalico.org/v1/ippool.go

```diff
@@ -77,13 +110,29 @@ const (
 	IPIPModeCrossSubnet IPIPMode = "CrossSubnet"
 )
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// The following definitions are only used for APIv1 backwards compatibility.
+// They are for internal use only.
+type EncapMode string
 
-// IPPoolList contains a list of IPPool resources.
-type IPPoolList struct {
-	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata"`
-	Items           []IPPool `json:"items"`
+const (
+	Undefined   EncapMode = ""
+	Always      EncapMode = "always"
+	CrossSubnet EncapMode = "cross-subnet"
```

Comment:
These don't seem to be used either?

Reply:
(Just on IPIPConfiguration, which is an outdated struct)

Reply:
I removed all the unneeded changes.